caktus_primer/static/js/modal.js

```diff
@@ -0,0 +1,69 @@
+let clickEvents = ['click', 'mousedown', 'touchstart', 'dblclick']
+
+
+class Modal {
+    constructor(id) {
+        this.name = 'Modal'
+        this.options = {}
+        this.container = document.getElementById(id) || null
+        if (this.container) {
+            this.openTarget = this.container.querySelector('[aria-label="open"]')
+            this.dialogTarget = this.container.getElementsByTagName('dialog')[0]
+            this.closeTarget = this.container.querySelector('[aria-label="close"]')
+        }
+        else {
+            return
+        }
+
+        // Attach event listeners to our click events to handle opening and closing the modal window
+        clickEvents.forEach((n) => {
+            this.openTarget.addEventListener(n, (e) => { this.handleClick(e) })
+            this.closeTarget.addEventListener(n, (e) => { this.closeModal(e) })
+        })
+        
+        document.addEventListener('keypress', (e) => { 
+            // Escape key
+            if (e.keyCode == 27) {
+                this.closeModal(e) 
+            }
+        })
```

Comment:
Lets put these bits up in that if statement, and then we can get rid of the else and early-return. It might make the constructor easier to read if there isn't that early return to overlook.